admin/asset_files/forms.py

```diff
@@ -7,6 +7,15 @@
 class ProviderAssetFileForm(forms.ModelForm):
     class Meta:
         model = ProviderAssetFile
-        fields = ['name', 'file', 'providers']
+        fields = ['name', 'file', 'providers', 'id']
 
+    id = forms.IntegerField(required=False, widget=forms.HiddenInput())
     providers = forms.ModelMultipleChoiceField(AbstractProvider.objects.all(), widget=forms.CheckboxSelectMultiple(), required=False)
+
+    def clean(self):
+        cleaned_data = super(ProviderAssetFileForm, self).clean()
+        obj_id = int(cleaned_data.get('id', None) or 0)
+        for provider in cleaned_data.get('providers', []):
+            if provider.asset_files.exclude(id=obj_id).filter(name=cleaned_data.get('name', '')).exists():
```

Comment:
Can they have no name? (the '' default case for name). If not, should something else raise or at least bypass hitting the DB?